flaml/automl/automl.py

```diff
@@ -2560,12 +2614,49 @@ def cv_score_agg_func(val_loss_folds, log_metrics_folds):
         )
         min_sample_size = min_sample_size or self._settings.get("min_sample_size")
         use_ray = self._settings.get("use_ray") if use_ray is None else use_ray
+        use_spark = self._settings.get("use_spark") if use_spark is None else use_spark
+        if use_spark and use_ray is not False:
+            raise ValueError("use_spark and use_ray cannot be both True.")
+        elif use_spark:
+            check_spark()
+
+        old_level = logger.getEffectiveLevel()
+        self.verbose = verbose
+        logger.setLevel(50 - verbose * 10)
+        if not logger.handlers:
+            # Add the console handler.
+            _ch = logging.StreamHandler(stream=sys.stdout)
+            _ch.setFormatter(logger_formatter)
+            logger.addHandler(_ch)
+
+        if not use_ray and not use_spark and n_concurrent_trials > 1:
+            if ray_import:
+                logger.warning(
+                    "n_concurrent_trials > 1 is only supported when using Ray or Spark. "
+                    "Ray installed, setting use_ray to True. If you want to use Spark, set use_spark to True."
+                )
+                use_ray = True
+            elif spark_import:
+                logger.warning(
+                    "n_concurrent_trials > 1 is only supported when using Ray or Spark. "
+                    "Spark installed, setting use_spark to True. If you want to use Ray, set use_ray to True."
+                )
+                use_spark = True
+                check_spark()
+            else:
+                logger.warning(
+                    "n_concurrent_trials > 1 is only supported when using Ray or Spark. "
+                    "Neither Ray nor Spark installed, setting n_concurrent_trials to 1."
+                )
+                n_concurrent_trials = 1
```

Comment:
Do we just want to throw here? Feels like misconfiguration to ask for concurrent trials with no parallel executor

Reply:
The reason I just log a warning here is to make flaml more friendly to users. The same config can run in different environments.

Reply:
#860 created in case we want to change the behavior in future.